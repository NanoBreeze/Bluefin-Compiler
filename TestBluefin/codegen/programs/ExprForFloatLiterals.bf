/*
Verifies that basic expressions for float literals are okay
No type promotions here. All types are float. 
We use nice rounded float numbers because their floating point impl is nicer for the eye
*/

void foo(float val) {

	1.0+2.1*3.3; // expect constant folding so no IR generated
	val+5.0 - 9.5*val - val/3.5; 
	
	val > 5.0;
	val >= 5.0;
	val < 5.0;
	val <= 5.0;

	val == 5.0;
	val != 5.0;
	-val;

	(val+1.0)*2.5;

}
