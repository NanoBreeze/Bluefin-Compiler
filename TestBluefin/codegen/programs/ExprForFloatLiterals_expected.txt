; ModuleID = 'ExprForFloatLiterals.bf'
source_filename = "ExprForFloatLiterals.bf"

define void @foo(float %val) {
entry:
  %alloctmp_val = alloca float, align 4
  store float %val, float* %alloctmp_val, align 4
  %loadtmp_val = load float, float* %alloctmp_val, align 4
  %addtmp = fadd float %loadtmp_val, 5.000000e+00
  %loadtmp_val1 = load float, float* %alloctmp_val, align 4
  %multmp = fmul float 9.500000e+00, %loadtmp_val1
  %subtmp = fsub float %addtmp, %multmp
  %loadtmp_val2 = load float, float* %alloctmp_val, align 4
  %divtmp = fdiv float %loadtmp_val2, 3.500000e+00
  %subtmp3 = fsub float %subtmp, %divtmp
  %loadtmp_val4 = load float, float* %alloctmp_val, align 4
  %cmpFGTtmp = fcmp ogt float %loadtmp_val4, 5.000000e+00
  %loadtmp_val5 = load float, float* %alloctmp_val, align 4
  %cmpFGEtmp = fcmp oge float %loadtmp_val5, 5.000000e+00
  %loadtmp_val6 = load float, float* %alloctmp_val, align 4
  %cmpFLTtmp = fcmp olt float %loadtmp_val6, 5.000000e+00
  %loadtmp_val7 = load float, float* %alloctmp_val, align 4
  %cmpFLEtmp = fcmp ole float %loadtmp_val7, 5.000000e+00
  %loadtmp_val8 = load float, float* %alloctmp_val, align 4
  %cmpEQtmp = fcmp oeq float %loadtmp_val8, 5.000000e+00
  %loadtmp_val9 = load float, float* %alloctmp_val, align 4
  %cmpNEtmp = fcmp one float %loadtmp_val9, 5.000000e+00
  %loadtmp_val10 = load float, float* %alloctmp_val, align 4
  %negtmp = fneg float %loadtmp_val10
  %loadtmp_val11 = load float, float* %alloctmp_val, align 4
  %addtmp12 = fadd float %loadtmp_val11, 1.000000e+00
  %multmp13 = fmul float %addtmp12, 2.500000e+00
  ret void
}
